executor/set_test.go

```diff
@@ -601,6 +601,16 @@ func TestSetVar(t *testing.T) {
 	tk.MustQuery("select @@global.tidb_remove_orderby_in_subquery").Check(testkit.Rows("0")) // default value is 0
 	tk.MustExec("set global tidb_remove_orderby_in_subquery=1")
 	tk.MustQuery("select @@global.tidb_remove_orderby_in_subquery").Check(testkit.Rows("1"))
+
+	// the value of max_allowed_packet should be a multiple of 1024
+	tk.MustExec("set @@global.max_allowed_packet=16385")
+	tk.MustQuery("show warnings").Check(testkit.RowsWithSep("|", "Warning|1292|Truncated incorrect max_allowed_packet value: '16385'"))
+	result := tk.MustQuery("select @@global.max_allowed_packet;")
+	result.Check(testkit.Rows("16384"))
+	tk.MustExec("set @@max_allowed_packet=2047")
+	tk.MustQuery("show warnings").Check(testkit.RowsWithSep("|", "Warning|1292|Truncated incorrect max_allowed_packet value: '2047'"))
+	result = tk.MustQuery("select @@max_allowed_packet;")
+	result.Check(testkit.Rows("1024"))
```

Comment:
Please add a test case for `set @@global.max_allowed_packet=0`(the result should be `1024`).

Also, I noticed that MySQL gives the following warning for `set @@global.max_allowed_packet=0`:

```
mysql> show warnings;
+---------+------+-------------------------------------------------------------------------------------------+
| Level   | Code | Message                                                                                   |
+---------+------+-------------------------------------------------------------------------------------------+
| Warning | 1292 | Truncated incorrect max_allowed_packet value: '0'                                         |
| Warning | 1708 | The value of 'max_allowed_packet' should be no less than the value of 'net_buffer_length' |
+---------+------+-------------------------------------------------------------------------------------------+
```
However, `net_buffer_length` is a no-op variable in TiDB, so I don't think this warning is necessary so we don't have to add the extra warning check for it. /cc @morgo

Reply:
I agree it's not necessary to add the warning.